### PR TITLE
Add MongoSpy unit test and move original test to integration

### LINKED ISSUE #581: Split MongoSpy tests into integration and unit
The current MongoSpy tests start/stop a mongodb server, so take quite a bit of time.
We should split the tests into unit and integration.
Integration tests should use a real MongoDB server and unit tests should not require MongoDB to be running.

### PR DESCRIPTION
**Pull request by estolfo**
Closes elastic/apm-agent-ruby#581

### CODE REVIEW

**Comment by estolfo** [comment]
@mikker Would you like me to add a flag so that the mongo integration tests conditionally run?

**Comment by mikker** [comment]
Sure, yes, I think a flag would be good. Otherwise having both kind of disputes the purpose of making the suite faster.

1. Check for `ENV['CI']`
1. Check for `ENV['RUN_FULL_SUITE']` or something else we define now
1. Check for `END['INCLUDE_MONGO_SPEC']` or similar

Which one do you prefer?

**Comment by estolfo** [comment]
Let's start with the simplest: `ENV['CI']`. (and I'm assuming this is set in Jenkins)
If we need to further categorize our specs later, we can add a separate env variable. I've just commit a check for `ENV['CI']`

**Comment by estolfo** [comment]
@mikker I've updated this to fit with the recent test changes. Let me know if you're ok with it and I'll merge.

File: spec/integration/mongo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    after(:context) do
+      stop_mongodb
+    end
+
+    def stop_mongodb
+      `docker-compose -f spec/docker-compose.yml down -v 2>&1`
+    end
+
+    def start_mongodb
```
Comment (mikker) [inline spec/integration/mongo_spec.rb]: With the new dev setup we can just rely on this being true 😅

**Pull request merged**